pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -442,22 +469,30 @@ func (w *wlReconciler) nominateAndSynchronizeWorkers(ctx context.Context, group
 	log.V(3).Info("Nominate and Synchronize Worker Clusters")
 	var nominatedWorkers []string
 
-	if w.dispatcherName == config.MultiKueueDispatcherModeAllAtOnce {
-		for workerName := range group.remotes {
-			nominatedWorkers = append(nominatedWorkers, workerName)
-		}
-		if group.local.Status.ClusterName == nil && !equality.Semantic.DeepEqual(group.local.Status.NominatedClusterNames, nominatedWorkers) {
-			if err := workload.PatchAdmissionStatus(ctx, w.client, group.local, w.clock, func() (*kueue.Workload, bool, error) {
-				group.local.Status.NominatedClusterNames = nominatedWorkers
-				return group.local, true, nil
-			}); err != nil {
-				log.V(2).Error(err, "Failed to patch nominated clusters", "workload", klog.KObj(group.local))
-				return reconcile.Result{}, err
+	// For elastic workloads, retrieve the remote cluster where the original workload was scheduled.
+	// For now, new workload slices will continue to be assigned to the same cluster.
+	// In the future, we may introduce more nuanced remote workload propagation policies,
+	// supporting preferred or required placement constraints.
+	if clusterName := workload.ClusterName(group.local); group.IsElasticWorkload() && clusterName != "" {
+		nominatedWorkers = []string{clusterName}
+	} else {
+		if w.dispatcherName == config.MultiKueueDispatcherModeAllAtOnce {
```

Comment:
I would prefer else-if or "switch" just to avoid indentation. On the first review pass seeing the last diff I was tricked to think the dispatchers logic is changed. Also, saving the indentation level is a win for readability on its own.